circuit/environment/src/helpers/r1cs.rs

```diff
@@ -19,18 +19,18 @@ use crate::{
 };
 use snarkvm_fields::PrimeField;
 
-use std::rc::Rc;
+use std::sync::Arc;
 
 pub type Scope = String;
 
 #[derive(Debug)]
 pub struct R1CS<F: PrimeField> {
     constants: Vec<Variable<F>>,
-    public: Vec<Variable<F>>,
-    private: Vec<Variable<F>>,
-    constraints: Vec<Rc<Constraint<F>>>,
+    pub(crate) public: Vec<Variable<F>>,
+    pub(crate) private: Vec<Variable<F>>,
+    pub(crate) constraints: Vec<Arc<Constraint<F>>>,
```

Comment:
Can you make a tracking issue for remaining future synthesis speedup ideas? 
- You can note that you didn't find a speedup with arenas.
- I suspect we might be able to hide some of this tracking in `R1CS` and `Counter` behind a feature flag. (applies to `CheckSynthesis` benchmark)
- Maybe we can get rid of the intermediary `Converter` and new `LinearCombination` in `generate_constraints` (applies to `verify_vk` benchmark)
- Maybe we can pass in `Assignment` by value. (applies to `verify_vk` benchmark)